test/e2e/metering_manual_install_tests.go

```diff
@@ -104,6 +106,75 @@ func testManualMeteringInstall(
 	assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
 }
 
+func DecodeMeteringConfigManifest(basePath, manifestPath, manifestFilename string) (*metering.MeteringConfig, error) {
+	manifestFullPath := filepath.Join(basePath, manifestPath, manifestFilename)
+	file, err := os.Open(manifestFullPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open the %s manifest file: %v", manifestFullPath, err)
+	}
+
+	mc := &metering.MeteringConfig{}
+	err = yaml.NewYAMLOrJSONDecoder(file, 100).Decode(&mc)
+	if err != nil {
+		return nil, err
+	}
+
+	if mc == nil {
+		return nil, fmt.Errorf("error: the decoded MeteringConfig object is nil")
+	}
+
+	return mc, nil
+}
+
+func createNFSProvisioner(ctx *deployframework.DeployerCtx) error {
+// cteate namespace
+ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+_, err := ctx.Client.CoreV1().Namespaces().Get(context.Background(), ctx.Namespace, metav1.GetOptions{})
+if err != nil && !apierrors.IsNotFound(err) {
+	return nil
+}
+if apierrors.IsNotFound(err) {
+	n := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ctx.Namespace,
+			Labels: map[string]string{
+				"name": ctx.Namespace + "-" + testingNamespaceLabel,
+			},
+		},
+	}
+	_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), n, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	ctx.Logger.Debugf("Created the %s namespace", ctx.Namespace)
+}
+//decode and create yamls except for persistentvolume
+filepathNFSServer := "../testhelpers/nfs_server.yaml"
+filepathNFSService := "../testhelpers/nfs_service.yaml"
+filepathNFSStorageClass := "../testhelpers/nfs_storageclass.yaml"
+filepathNFSPersistentvolume := "../testhelpers/nfs_persistentvolume.yaml"
+
+err := deploy.DecodeYAMLManifestToObject(filepathNFSServer, &nfsserver)
+if err != nil {
+	return "", fmt.Errorf("failed to decode the %s YAML service manifest: %v", filepathNFSServer, err)
+}
+err := deploy.DecodeYAMLManifestToObject(filepathNFSService, &nfsservice)
+if err != nil {
+	return "", fmt.Errorf("failed to decode the %s YAML service manifest: %v", filepathNFSService, err)
+}
+err := deploy.DecodeYAMLManifestToObject(filepathNFSStorageClass, &service)
+if err != nil {
+	return "", fmt.Errorf("failed to decode the %s YAML service manifest: %v", filepathNFSStorageClass, err)
+}
+//  - Wait until the Service created has a populated spec.ClusterIP (poll.Wait functionality)
+//  - Decode the NFS PV YAML manifest
+
+err := deploy.DecodeYAMLManifestToObject(filepathNFSPersistentvolume, &nfs_persistentvolume)
+if err != nil {
+	return "", fmt.Errorf("failed to decode the %s YAML service manifest: %v", filepathNFSPersistentvolume, err)
+}
+
```

Comment:
This isn't going to compile.

Reply:
Yeah, still WIP, the clusterIP polling has to occur too. Was mostly wondering about the use of `deploy.DecodeYAMLManifestToObject` with multiple filepaths.